### PR TITLE
graph: support admin consent

### PR DESCRIPTION
**Pull request by yugangw-msft**
With this, CLI will do exactly like the "Grant Permission" button in portal and hence both "Application" and "Delegated" permissions can be handled.

This addresses the lingering ask in #7177 and #8310 

//CC @paolosalvatori @jianghaolu 

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: src/command_modules/azure-cli-role/azure/cli/command_modules/role/custom.py [hunk 1]
```diff
@@ -807,6 +807,34 @@ def delete_permission(cmd, identifier, api):
...
 
 
+def admin_consent(cmd, identifier):
+    import requests
+    from azure.cli.core.cloud import AZURE_PUBLIC_CLOUD
+    from azure.cli.core._profile import Profile
+    from azure.cli.core.commands.client_factory import UA_AGENT
+    from azure.cli.core.util import should_disable_connection_verify
+    if cmd.cli_ctx.cloud.name != AZURE_PUBLIC_CLOUD.name:
+        raise CLIError('This command does not function yet in solverign clouds')
```
Comment (marstr) [inline src/command_modules/azure-cli-role/azure/cli/command_modules/role/custom.py]: typo! "solverign" should be "sovereign"

nit, I'd rephrase this to "This command is not yet supported on sovereign clouds"

File: src/command_modules/azure-cli-role/azure/cli/command_modules/role/custom.py [hunk 2]
```diff
@@ -807,6 +807,34 @@ def delete_permission(cmd, identifier, api):
     return graph_client.applications.patch(application.object_id, update_parameter)
 
 
+def admin_consent(cmd, identifier):
+    import requests
+    from azure.cli.core.cloud import AZURE_PUBLIC_CLOUD
+    from azure.cli.core._profile import Profile
```
Comment (marstr) [inline src/command_modules/azure-cli-role/azure/cli/command_modules/role/custom.py]: Importing from private namespaces always raises my eyebrows. Is `Profile` something we should bring into the public interface?

File: src/command_modules/azure-cli-role/azure/cli/command_modules/role/_help.py [hunk 3]
```diff
@@ -200,11 +200,17 @@
...
+    short-summary: Grant the app an API Delegated permissions
+    long-summary: for Application permissions, please use "ad app permission admin-consent"
     examples:
         - name: Grant a native application with permissions to access an existing API with TTL of 2 years
           text: az ad app permission grant --id e042ec79-34cd-498f-9d9f-1234234 --api a0322f79-57df-498f-9d9f-12678 --expires 2
 """
+helps['ad app permission admin-consent'] = """
+    type: command
+    short-summary: grant Application & Delegated permissions through admin-consent.
+    long-summary: you must login as a directory administrator
```
Comment (marstr) [inline src/command_modules/azure-cli-role/azure/cli/command_modules/role/_help.py]: This seems like a good detail to include. When we present help text, do we always present the short summary when we present the long summary? If not, I think we should repeat the short summary there as well.

Comment (yugangw-msft) [reply src/command_modules/azure-cli-role/azure/cli/command_modules/role/_help.py]: Long summary will always be present from command's help. It will not be visible from a category's help such as `az ad permission -h`.
This should be good enough

File: src/command_modules/azure-cli-role/azure/cli/command_modules/role/custom.py (hunk 2 above)
Comment (yugangw-msft) [reply src/command_modules/azure-cli-role/azure/cli/command_modules/role/custom.py]: `_profile` is okay to use from core command modules which `role`, `keyvault` command modules have done that already, because we are always releasing them together.
